src/formats/roam-json.ts

```diff
@@ -266,10 +357,56 @@ export class RoamJSONImporter extends FormatImporter {
 
 		if (json.children) {
 			for (const child of json.children) {
-				markdown.push(await this.jsonToMarkdown(graphFolder, attachmentsFolder, child, indent + '  ', true));
+				markdown.push(await this.jsonToMarkdown(graphFolder, attachmentsFolder, child, indent + '  ', true, checkYAMLoptions, false, '', this.oldestTimestamp, this.newestTimestamp));
+			}
+		}
+
+		// once processing children is completed, add the YAML to the top
+		// check if any YAML options are set
+		switch (!checkYAMLoptions) {
+			case this.fileDateYAML:
+			case this.titleYAML:
+			// add other options here
+			// case this.{option-name}:
+				checkYAMLoptions = true;
+		}
+
+		// add YAML frontmatter, if enabled
+		if (checkYAMLoptions === true && runYAMLoptions === true) {
+
+			let timeCreated = this.oldestTimestamp;
+
+			frontMatterYAML.push('---');
+
+			// if "add title" option enabled, quotes added to prevent errors in frontmatter
+			if (this.titleYAML === true) {
+				frontMatterYAML.push('title: "' + setTitleProperty +'"');
+			}
+
+			// if "timestamps" option enabled
+			if (this.fileDateYAML === true) {
+				// use Roam's create-time and edit-time values to set timestamps
+				// if create is missing, use updated
+				// if updated is missing, use current Date()
+				let TSFormat = "YYYY-MM-DD HH:mm:ss";
+			
+				let formatUpdateDate = this.newestTimestamp ? moment(this.newestTimestamp).format(TSFormat) : moment(new Date()).format(TSFormat);
+				let formatCreateDate = timeCreated ? moment(timeCreated).format(TSFormat) : formatUpdateDate;
+
+				frontMatterYAML.push('created: ' + formatCreateDate);
+				frontMatterYAML.push('updated: ' + formatUpdateDate);
 			}
+
+			frontMatterYAML.push('---');
+			runYAMLoptions = false;
 		}
 
+		// Add frontmatter YAML to the top of the markdown array
+		// reverse it before pushing
+		frontMatterYAML.reverse();
```

Comment:
It would be much more efficient to join our string and just add one item to the markdown.

```
markdown.unshift(frontMatterYAML.join('\n'));
```